tests/template_tests/test_context.py

```diff
@@ -96,6 +110,17 @@ def test_flatten_context(self):
             'a': 2, 'b': 4, 'c': 8
         })
 
+    def test_flatten_context_with_context(self):
+        """
+        #24765 -- Verifies Using Context with Flatten Works
+        """
+        a = Context({'a': 2})
+        a.push(Context({'z': '8'}))
+
+        self.assertEqual(a.flatten(), {
+            'False': False, 'None': None, 'True': True, 'a': 2, 'z': '8'
```

Comment:
I'd put each key/value on its own line and please include a trailing comma on the last item (habit so that if more items are added later, we don't need to change that line again)